BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BluetoothBase.java

```diff
@@ -35,53 +31,58 @@ public BluetoothBase(Context Context, BluetoothStatusChanged handler) {
 
         //bluetooth = new BluetoothAdapter(this);
         bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null) {
+        if (bluetooth != null) {
             blueAddress = bluetooth.getAddress();
         }
     }
 
-    public boolean Start() {
+    public synchronized  boolean Start() {
```

Comment:
only bad thing that can happen there would be the registerReceiver throwing an error, and currently I'm catching that there and reporting that we are not started with returning false to the caller.